mantis-client/src/main/java/io/mantisrx/client/MantisClient.java

```diff
@@ -101,6 +102,9 @@ public MantisClient(Properties properties) {
         HighAvailabilityServices haServices =
             HighAvailabilityServicesUtil.createHAServices(
                 Configurations.frmProperties(properties, CoreConfiguration.class));
+        if (haServices.state() == State.NEW) {
+            haServices.startAsync().awaitRunning();
```

Comment:
Is it possible for the haServices not being started right now but started later?

Reply:
In that case, you should use the other constructor that allows you to control the lifecycle of the HighAvailabilityServices better.